core/src/main/java/cz/o2/proxima/scheme/SchemaDescriptors.java

```diff
@@ -0,0 +1,885 @@
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link ArrayTypeDescriptor} with name.
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(String name, ArrayTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link StructureTypeDescriptor} with name
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(
+      String name, StructureTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor.toTypeDescriptor());
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link SchemaTypeDescriptor} with name
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(String name, SchemaTypeDescriptor<T> descriptor) {
+    return fields(name, descriptor, null);
+  }
+
+  /**
+   * Create {@link FieldTypeDescriptor} of {@link SchemaTypeDescriptor} with name and default value.
+   *
+   * @param name name of field
+   * @param descriptor value descriptor
+   * @param defaultValueFactory default value factory
+   * @param <T> value type
+   * @return field descriptor
+   */
+  public static <T> FieldTypeDescriptor<T> fields(
+      String name, SchemaTypeDescriptor<T> descriptor, Factory<T> defaultValueFactory) {
+    return new FieldTypeDescriptor<>(name, descriptor, defaultValueFactory);
+  }
+
+  /**
+   * Generic type descriptor. Parent class for other types
+   *
+   * @param <T> value type
+   */
+  public abstract static class GenericTypeDescriptor<T> implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /** Value type */
+    @Getter protected final AttributeValueType type;
+
+    protected GenericTypeDescriptor(AttributeValueType type) {
+      this.type = type;
+    }
+
+    /**
+     * Convert specific type descriptor to {@link SchemaTypeDescriptor}
+     *
+     * @return descriptor
+     */
+    public abstract SchemaTypeDescriptor<T> toTypeDescriptor();
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o instanceof GenericTypeDescriptor) {
+        return type.equals(((GenericTypeDescriptor<?>) o).getType());
+      }
+      return false;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(type);
+    }
+  }
+
+  /**
+   * SchemaTypeDescriptor wrapper
+   *
+   * @param <T> value type
+   */
+  public static class SchemaTypeDescriptor<T> extends GenericTypeDescriptor<T> {
```

Comment:
I'm reading this quite fast, but the purpose of this class is a bit mystery to me.